waku/v2/node/builder.nim

```diff
@@ -145,21 +151,45 @@ proc build*(builder: WakuNodeBuilder): Result[WakuNode, string] =
   if builder.netConfig.isNone():
     return err("network configuration is required")
 
-  var node: WakuNode
+  if builder.maxRelayPeers.isNone():
+    return err("max relay peers is required")
+
+  var switch: Switch
   try:
-    node = WakuNode.new(
+    switch = newWakuSwitch(
+      privKey = builder.nodekey,
+      address = builder.netConfig.get().hostAddress,
+      wsAddress = builder.netConfig.get().wsHostAddress,
+      transportFlags = {ServerFlags.ReuseAddr},
       rng = rng,
-      nodeKey = builder.nodeKey.get(),
-      netConfig = builder.netConfig.get(),
-      peerStorage = builder.peerStorage.get(nil),
-      peerStoreCapacity = builder.peerStorageCapacity,
       maxConnections = builder.switchMaxConnections.get(builders.MaxConnections),
+      wssEnabled = builder.netConfig.get().wssEnabled,
+      secureKeyPath = builder.switchSslSecureKey.get(""),
+      secureCertPath = builder.switchSslSecureCert.get(""),
       nameResolver = builder.switchNameResolver.get(nil),
-      agentString = builder.switchAgentString,
-      secureKey = builder.switchSslSecureKey.get(""),
-      secureCert = builder.switchSslSecureCert.get(""),
       sendSignedPeerRecord = builder.switchSendSignedPeerRecord.get(false),
+      agentString = builder.switchAgentString,
+      peerStoreCapacity = builder.peerStorageCapacity,
+      services = @[Service(getAutonatService(rng))],
+    )
+  except:
```

Comment:
I think we need to add a specific catch type to avoid warning :)
```suggestion
  except CatchableError:
```

Reply:
fixed 5bf87652a36b72f5d9429480bdbb538952b187e4